cpp/src/gandiva/function_registry.h

```diff
@@ -30,18 +35,41 @@ class GANDIVA_EXPORT FunctionRegistry {
  public:
   using iterator = const NativeFunction*;
 
+  FunctionRegistry();
+  FunctionRegistry(const FunctionRegistry&) = delete;
+  FunctionRegistry& operator=(const FunctionRegistry&) = delete;
+
   /// Lookup a pre-compiled function by its signature.
   const NativeFunction* LookupSignature(const FunctionSignature& signature) const;
 
+  /// \brief register a set of functions into the function registry from a given bitcode
+  /// file
+  arrow::Status Register(const std::vector<NativeFunction>& funcs,
+                         const std::string& bitcode_path);
+
+  /// \brief register a set of functions into the function registry from a given bitcode
+  /// buffer
+  arrow::Status Register(const std::vector<NativeFunction>& funcs,
+                         std::unique_ptr<arrow::Buffer> bitcode_buffer);
```

Comment:
This would be more flexible if it took a `std::shared_ptr<arrow::Buffer>`

Reply:
Sure. Updated to use `shared_ptr`, and the internal storage of `arrow::Buffer` becomes `std::vector<std::shared_ptr<arrow::Buffer>>` now.